Source/Core/Core/IOS/ES/NandUtils.cpp

```diff
@@ -163,6 +163,23 @@ std::vector<Content> GetStoredContentsFromTMD(const TMDReader& tmd)
   return stored_contents;
 }
 
+u32 GetSharedContentsCount()
+{
+  const std::string shared1_path = Common::RootUserPath(Common::FROM_SESSION_ROOT) + "/shared1";
+  const auto entries = File::ScanDirectoryTree(shared1_path, false);
+  return static_cast<u32>(
+      std::count_if(entries.children.begin(), entries.children.end(), [](const auto& entry) {
+        return !entry.isDirectory && entry.virtualName.size() == 12 &&
+               entry.virtualName.substr(8, 12) == ".app";
```

Comment:
You can likely use [std::string::compare](http://en.cppreference.com/w/cpp/string/basic_string/compare) here instead of creating a substring.